ResidentsSocialCarePlatformApi.Tests/V1/Gateways/SocialCare/GetVisitInformationByPersonIdTests.cs

```diff
@@ -0,0 +1,104 @@
+using ResidentsSocialCarePlatformApi.V1.Infrastructure;
+
+namespace ResidentsSocialCarePlatformApi.Tests.V1.Gateways.SocialCare
+{
+    [NonParallelizable]
+    [TestFixture]
+    public class GetVisitInformationByPersonIdTests : DatabaseTests
+    {
+        private SocialCareGateway _classUnderTest;
+
+        [SetUp]
+        public void Setup()
+        {
+            _classUnderTest = new SocialCareGateway(SocialCareContext);
+        }
+
+        [Test]
+        public void WhenThereAreNoVisitsWithPersonId_ReturnsEmptyList()
+        {
+            const long realPersonId = 123L;
+            const long fakePersonId = 456L;
+            AddVisitWithAPerson(personId: realPersonId);
+
+            var response = _classUnderTest.GetVisitInformationByPersonId(fakePersonId);
+
+            response.Should().BeEmpty();
+        }
+
+        [Test]
+        public void WhenThereIsOneMatch_ReturnsAListContainingTheMatchingCaseNote()
+        {
+            const long realPersonId = 123L;
+            var visit = AddVisitWithAPerson();
+
+            var response = _classUnderTest.GetVisitInformationByPersonId(realPersonId);
+
+            response.Count.Should().Be(1);
+            response.FirstOrDefault().VisitId.Should().Be(visit.VisitId);
+        }
+
+        [Test]
+        public void WhenThereAreMultipleMatches_ReturnsAListContainingAllMatchingCaseNotes()
+        {
+            const long realPersonId = 123L;
+            AddVisitWithAPerson(realPersonId);
+            AddVisitWithAPerson(realPersonId);
+
+            var response = _classUnderTest.GetVisitInformationByPersonId(realPersonId);
+
+            response.Count.Should().Be(2);
+        }
+
+        [Test]
+        public void OnlyReturnVisitsForCorrectPersonId()
+        {
+            const long realPersonId = 123L;
+            const long otherPersonId = 456L;
+            AddVisitWithAPerson(personId: realPersonId);
+            AddVisitWithAPerson(personId: otherPersonId);
+
+            var response = _classUnderTest.GetVisitInformationByPersonId(realPersonId);
+
+            response.Count.Should().Be(1);
+        }
+
+        [Test]
+        public void WhenThereIsAMatchingRecord_ReturnsDetailsFromVisit()
+        {
+            var visit = AddVisitWithAPerson();
+
+            var response = _classUnderTest.GetVisitInformationByPersonId(visit.PersonId).First();
+
+            response.VisitId.Should().Be(visit.VisitId);
+            response.PersonId.Should().Be(visit.PersonId);
+            response.VisitType.Should().Be(visit.VisitType);
+            response.PlannedDateTime.Should().Be(visit.PlannedDateTime);
+            response.ActualDateTime.Should().Be(visit.ActualDateTime);
+            response.ReasonVisitNotMade.Should().Be(visit.ReasonVisitNotMade);
+            response.SeenAloneFlag.Should().Be(visit.SeenAloneFlag);
+            response.CompletedFlag.Should().Be(visit.CompletedFlag);
+            response.OrgId.Should().Be(visit.OrgId);
```

Comment:
I think we should return the organisation name as part of the visit like with case notes we return `createdByName` from the `workers` table. The Org ID is only Mosaic specific which I don't think can be used by the service API and frontend.